simplebgc/src/commands/cmd_response.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{Payload, PayloadParseError};
+use bytes::{Buf, BufMut, Bytes, BytesMut};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct Confirm {
+    pub cmd_id: u8,
+    pub data: Option<DataType>,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DataType {
+    DataU8(u8),
+    DataU16(u16),
+}
+
+impl Payload for Confirm {
+    fn from_bytes(mut b: Bytes) -> Result<Self, PayloadParseError>
+    where
+        Self: Sized,
+    {
+        Ok(Confirm {
+            cmd_id: read_enum!(b, "CMD_ID", u8)?,
+            // For some reason I was observing behavior inconsistent with the docs where
+            // I was getting 0 bytes here when it says there should be 1 or 2 bytes.
+            data: if b.remaining() == 0 {
+                None
+            } else if b.remaining() == 1 {
+                Some(DataType::DataU8(read_enum!(b, "DATA", u8)?))
+            } else if b.remaining() == 2 {
+                Some(DataType::DataU16(read_enum!(b, "DATA", u16)?))
+            } else {
```

Comment:
Why does the end user care about whether the gimbal actually sent a `u8` or a `u16`? If we get a `u8`, can't we just cast it into a `u16`?

Reply:
Sure. I wrote this intending to stay true to the API spec so the user can cross-check with it but that's good too.